### DISCUSSION TITLE
What Is extra_trees=True?

### DISCUSSION THREAD

**Discussion by simonprovost** [autogluon/autogluon#1900] [Q&A] [answered]
Hi every1ne,

I was wondering whether you might tell me what is extra trees=true in for example LGBM models? Giving then the LGBMXT. Is it an ensemble of LGBM + extra tree from Scikit? IS it a known ensemble that AutoGluon implemented, or that comes from AutoGluon's authors? Furthermore, how can I duplicate this, let's say, AutoGluon output the LGBMXT as the best configuration (champion); I am using the LGB library, but how to I set the "extra tree", and when (prior or after), please? 

Additionally; I am requesting the leaderboard to output the hyper parameters chosen by the pipeline, and for LGBM, the following shows loads of parameters, I guess just the first object is significant for my situation of creating this champion model by hand, correct ? (i.e., learning rate and extra tree, the rest is unnecessary):

```
{'learning_rate': 0.05, 'extra_trees': True}  
{'num_boost_round': 10}
 {
  'max_memory_usage_ratio': 1.0, 
  'max_time_limit_ratio': 1.0,
  'max_time_limit': None,
  'min_time_limit': 0, 
  'valid_raw_types': ['bool', 'int', 'float', 'category'], 
  'valid_special_types': None,
  'ignored_type_group_special': None,
  'ignored_type_group_raw': None,
  'get_features_kwargs': None,
  'get_features_kwargs_extra': None,
  'predict_1_batch_size': None,
  'num_gpus': 1
}
```

Cheers;

**Comment by Innixma** [comment, marked as answer]
`extra_trees=True` is simply a hyperparameter in LightGBM, it is nothing unique to AutoGluon.

Yes, for your own recreation only the learning rate and extra_trees=True are necessary (as well as num_boost_round). Note that AutoGluon does various optimizations and tricks, and thus AutoGluon's LightGBM will often outperform a hand-made LightGBM, so you might consider using AutoGluon's LightGBM directly rather than trying to reproduce it stand-alone.

For details on available hyperparameters in LightGBM, you can view their docs: https://github.com/Microsoft/LightGBM/blob/master/docs/Parameters.rst

**Reply by simonprovost** [reply]
Crystal Clear @Innixma Cheers for his! I wasn't looking at the correct LGB doc earlier in the day. However, I got you on the AutoGluon's LightGBM internal implementation.
